### ISSUE TITLE
Suggestion about kernels for gp_optimize

### ISSUE THREAD

**Issue by cranmer** [scikit-optimize/scikit-optimize#399]
I've had a good first experience with skopt!

One thing tha that came up was tuning of the kernel for gp opt. I realize you can pass in your own GP, but a few suggestions on how the default. Currently you have:
https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/optimizer/gp.py#L227

`    if base_estimator is None:
        cov_amplitude = ConstantKernel(1.0, (0.01, 1000.0))

        if is_cat:
            other_kernel = HammingKernel(
                length_scale=np.ones(space.transformed_n_dims))
            acq_optimizer = "sampling"
        else:
            other_kernel = Matern(
                length_scale=np.ones(space.transformed_n_dims),
                length_scale_bounds=[(0.01, 100)] * space.transformed_n_dims,
                nu=2.5)

    base_estimator = GaussianProcessRegressor(
        kernel=cov_amplitude * other_kernel,
        normalize_y=True, random_state=rng, alpha=0.0, noise=noise,
        n_restarts_optimizer=2)`

I'm thinking that having the default values / range for the length_scale  be normalized to the range of the parameters would be a good idea. 

Also, in addition to the `noise` argument it could be useful to have an argument that specifies roughly the variability of the function value in the domain of the search. This is provided by the ConstantKernel now. Depending on the application 0.01 might not be small and 1000 might not be large and 1 might not be a good starting point. Maybe the user could provide something like `fcn_small` and `fcn_large`  and `ConstantKernel( (small+large)/2, (0.01*(large-small), 1000.0*(large-small))`.

**Comment by MechCoder** [comment]
> I've had a good first experience with skopt!

Great, thanks!

> I'm thinking that having the default values / range for the length_scale be normalized to the range of the parameters would be a good idea.

The parameters are normalized to the (0, 1) range [here](https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/optimizer/gp.py#L222). Also, beyond that the length_scale is tuned internally in sklearn so as to maximize the log-likelihood of the data, the values that you supply are just the initial values. (This has been a source of confusion multiple times, see: https://github.com/scikit-learn/scikit-learn/issues/8358

**Comment by MechCoder** [comment]
I think my comment also applies to your "variability-in-function" suggestion, In general, you can inspect the kernel parameters by doing `gp.kernel_.get_params()` after model fitting.

**Comment by cranmer** [comment]
I realize the GP optimizes those hyper parameters, but I didn't realize it was doing this rescaling. 

*for length scale*: 
We had to tune the GP kernel to get good results (however that was using the ask/tell API as in this:
https://github.com/cranmer/active_sciencing/blob/reintegration/bayesopt.py 
I'm not sure if the normalization of the parameters to (0,1) applies there or if we would have to do it by hand. Maybe that's why we had to tune it.

*for "variability in function* I also see that you can inspect the parameter of the kernel and provide a custom kernel. Maybe that's enough, but it does seem that the user can provide a generic hint as to roughly the range of the variability of the function without them having good insight into how to construct an appropriate GP kernel given that information. However, I also understand you might not want to add this complication to the API.

**Comment by MechCoder** [comment]
Whoops, nice catch! I opened https://github.com/scikit-optimize/scikit-optimize/issues/400 to track that.
